dht/range_streamer.cc

```diff
@@ -310,8 +311,12 @@ future<> range_streamer::do_stream_async() {
                             description, source, keyspace, nr_ranges_streamed, nr_ranges_total, ranges_to_stream.size());
                     nr_ranges_streamed += ranges_to_stream.size();
                     if (_nr_rx_added) {
+                        // This node is the stream receiver, trigger compaction on local node only
+                        _trigger_compaction[keyspace].insert(_address);
                         sp.request_ranges(source, keyspace, ranges_to_stream);
                     } else if (_nr_tx_added) {
+                        // This node is the stream sender, trigger compaction on remote node.
+                        _trigger_compaction[keyspace].insert(source);
```

Comment:
Why do you need to trigger compactions on the receiver from the sender? 
Wouldn't the branch above it be enough?

Reply:
Say n1, n2 in the cluster, n3 is bootstrapping, it receives from n1 and n2. n3 needs to trigger the compaction after n1 and n2 finishes streaming.  The first branch.

Say n1, n2, n3 in the cluster. n3 is decommissioning. When n3 finishes the ops, it needs to tell n1 and n2 to trigger the compaction. The second branch. 

Note range_streamer can either send or receive. It does not do both.

Reply:
I see. 
But what if the decommissioning node dies in the process?
I think you need to connect the compaction triggering to the same logic that you have today for deciding when streaming is over even if STREAMING_DONE is not received.